kupfer/plugin/websearch.py

```diff
@@ -18,7 +18,7 @@
 from kupfer.objects import TextLeaf
 from kupfer import utils, config
 
-from kupfer.plugin import firefox_support
+from kupfer.plugin import firefox
```

Comment:
Firefox seems to store available OpenSearch descriptions in a compressed config file now: <profile>/search.json.mozlz4. There is some code available for decompression that would have to be included into a (new) kupfer plugin. The resulting JSON contains names, urls for the OpenSearch XMLs it was generated from, the interactive search forms, the url templates for automatic searches and base64-encoded images, that may be the reason for a compression scheme being used. I assume Firefox updates its JSON metadata sometimes from the stored XML source urls, but probably not every time a search is done. So it seems safe to grab the names, url templates and maybe even icons from the cached JSON for kupfer integration.